### PR TITLE
update contribute/style

### PR DESCRIPTION
**Pull request by sgouezel**
Based on a discussion with @fpvandoorn at https://github.com/leanprover-community/mathlib/pull/5414#discussion_r550806210, I have the impression that the style guide does not completely reflect our current style. I propose some modifications in the current PR based on my feeling of what these practices are. Feedback/improvements/clarifications are most welcome!

### CODE REVIEW

File: templates/contribute/style.md [hunk 1]
```diff
@@ -20,6 +20,9 @@ rather than rigid rules.
 - `m`, `n`, `k`, ... for natural numbers
 - `i`, `j`, `k`, ... for integers
 
+Types with a mathematical content are expressed with the usual
+mathematical notation, often with an upper case letter 
+(`G` for a group, `R` for a ring, `K` or `𝕜` for a field, `E`, for a vector space, ...)
```
Comment (sgouezel) [inline templates/contribute/style.md:25]:
```suggestion
(`G` for a group, `R` for a ring, `K` or `𝕜` for a field, `E` for a vector space, ...)
```

File: templates/contribute/style.md [hunk 2]
```diff
@@ -400,21 +400,29 @@ If you are using multiple tactics inside a `by ...` block, use braces
...
 
-  definition foo (x : α) : α := x
+Very short goals can be closed right away using `swap` or `work_on_goal` if needed, to avoid
+additional indentation in the rest of the proof.
 
-  theorem bar (H : P) : P := H
-end my_section
+```lean
+begin
+  rw [h], swap, { exact h'},
```
Comment (sgouezel) [inline templates/contribute/style.md:422]:
```suggestion
  rw [h], swap, { exact h' },
```

**Comment by kbuzzard** [comment]
These changes look great to me. In particular the comment that lower case greek letters should be used for "generic types" (but sensible maths capital letters can be used for sensible maths objects) is long overdue and I think fits very well with what is happening right now (and is also a good system!)

**Comment by PatrickMassot** [comment]
LGTM.

File: templates/contribute/style.md [hunk 3]
```diff
@@ -10,7 +10,7 @@ rather than rigid rules.
 ### Variable conventions ###
 
 - `u`, `v`, `w`, ... for universes
-- `α`, `β`, `γ`, ... for types
+- `α`, `β`, `γ`, ... for generic types
```
Comment (sgouezel) [inline templates/contribute/style.md:13]:
```suggestion
- `α`, `β`, `γ`, ... for generic types
```

**Review by bryangingechen — APPROVED** [review]
Looks good to me too!

**Comment by fpvandoorn** [comment]
As mentioned in the Zulip thread, I would prefer to not have the convention "a single tactic invocation per line". The rest LGTM.

File: templates/contribute/style.md [hunk 4]
```diff
@@ -20,6 +20,9 @@ rather than rigid rules.
 - `m`, `n`, `k`, ... for natural numbers
 - `i`, `j`, `k`, ... for integers
 
+Types with a mathematical content are expressed with the usual
+mathematical notation, often with an upper case letter 
+(`G` for a group, `R` for a ring, `K` or `𝕜` for a field, `E` for a vector space, ...)
```
Comment (urkud) [inline templates/contribute/style.md:25]: Say that earlier we used `α`, `β` for groups as well, and PRs renaming all variables in a file are welcome?

**Comment by urkud** [comment]
Should we change "We generally restrict the use of dots to inductive types." in `naming.md` as well?

File: templates/contribute/style.md (hunk 4 above)
Comment (sgouezel) [reply templates/contribute/style.md]: Done

**Comment by sgouezel** [comment]
I have also made a few changes to `naming.md` as recommended by @urkud

**Review by Vierkantor — APPROVED** [review]
Looks good, thank you!

**Review by fpvandoorn — APPROVED** [review]
Looks good to me too. I'll merge it.

**Pull request merged**